packages/remix-react/data.ts

```diff
@@ -53,47 +61,7 @@ export async function fetchData(
     return error;
   }
 
+  // TODO: Confirm difference between regex extractData JSON detection versus
+  // @remix-run/router detection
   return response;
 }
-
-export async function extractData(response: Response): Promise<AppData> {
-  // This same algorithm is used on the server to interpret load
-  // results when we render the HTML page.
-  let contentType = response.headers.get("Content-Type");
-
-  if (contentType && /\bapplication\/json\b/.test(contentType)) {
```

Comment:
For the above TODO - in the router we use the following check - any concerns?

```
if (contentType && contentType.startsWith("application/json")) {
```

Reply:
Any reason to do a case insensitive check?

Reply:
Seems we do want to regex due to https://httpwg.org/specs/rfc9110.html#field.content-type